### PR TITLE
Web programming contribution

### PR DESCRIPTION
**Pull request by niranjanhegde144**
### **Describe your change:**
Added Currency Converter Program to web_programming directory. 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by niranjanhegde144** [comment]
Clauss Can you take a look at this

File: web_programming/currency_converter.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    apiKey: str = API_KEY,
+) -> str:
+    """https://www.amdoren.com/currency-api/"""
+    res = requests.get(
+        f"{URL_BASE}?api_key={API_KEY}&from={baseCurrency}&to={targetCurrency}&\
+            amount={amount}"
+    ).json()
+    if res["error"] == 0:
+        return str(res["amount"])
+    return res["error_message"]
```
Comment (cclauss) [inline web_programming/currency_converter.py:186]:
```suggestion
def convert_currency(
    from: str = "USD",
    to: str = "INR",
    amount: float = 1.0,
    api_key: str = API_KEY
) -> str:
    """https://www.amdoren.com/currency-api/"""
    res = requests.get(URL_BASE, data=locals()).json()
    return str(res["amount"] if res["error"] == 0 else res["error_message"])
```
There are nice advantages in requests and locals() if our function parameter names exactly match the API.

File: web_programming/currency_converter.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+if __name__ == "__main__":
+    base_currency = input("Enter base currency: ").strip()
+    target_currency = input("Enter target currency: ").strip()
+    amount = float(input("Enter the amount: ").strip())
+    print(
+        convert_currency(
+            baseCurrency=base_currency, targetCurrency=target_currency, amount=amount
+        )
+    )
```
Comment (cclauss) [inline web_programming/currency_converter.py:197]:
```suggestion
    print(
        convert_currency(
            input("Enter from currency: ").strip(),
            input("Enter to currency: ").strip(),
            float(input("Enter the amount: ").strip())
        )
    )
```

**Review by cclauss — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

**Review by cclauss — APPROVED** [review]

**Pull request merged**